src/lib/caches/Items.js

```diff
@@ -94,22 +100,74 @@ module.exports = class ItemsCache {
       let exact = false;
       filtered = undefined;
       base
-        .filter((item) => item && item[by])
+        .filter((item) => item && this.#parseNestedKey(item, bys))
         .forEach((item) => {
-          if (item[by].toLowerCase() === term.toLowerCase()) {
-            filtered = item;
-            exact = true;
+          const values = this.#parseNestedKey(item, bys);
+          // parseNestedKey returns an array, so check all values in that array
+          for (const entry of values) {
+            //logger.error(entry.toLowerCase());
+            if (entry.toString().toLowerCase() === term.toString().toLowerCase()) {
+              filtered = item;
+              exact = true;
+            }
           }
 
-          if (item[by].toLowerCase().includes(term.toLowerCase()) && !exact) {
-            const distance = item[by].toLowerCase().replace(term.toLowerCase(), '').length;
-            if (!keyDistance || distance < keyDistance) {
-              keyDistance = distance;
-              filtered = item;
+          if (!exact) {
+            // parseNestedKey returns an array, so check all values in that array
+            for (const entry of values.values()) {
+              if (entry.toString().toLowerCase().includes(term.toString().toLowerCase())) {
+                const distance = entry.toString().toLowerCase().replace(term.toString().toLowerCase(), '').length;
+                if (!keyDistance || distance < keyDistance) {
+                  keyDistance = distance;
+                  filtered = item;
+                }
+              }
             }
           }
```

Comment:
```suggestion
          if (!exact) {
            // parseNestedKey returns an array, so check all values in that array
            values.values().forEach((entry) => {
              if (entry.toString().toLowerCase().includes(term.toString().toLowerCase())) {
                const distance = entry.toString().toLowerCase().replace(term.toString().toLowerCase(), '').length;
                if (!keyDistance || distance < keyDistance) {
                  keyDistance = distance;
                  filtered = item;
                }
              }
            });
          }
```

Reply:
Converted to forEach() as suggested. Removed .values() call that should have been removed in previous version as it is not necessary.
```javascript
          if (!exact) {
            // parseNestedKey returns an array, so check all values in that array
            values.forEach((entry) => {
              if (entry.toString().toLowerCase().includes(term.toString().toLowerCase())) {
                const distance = entry.toString().toLowerCase().replace(term.toString().toLowerCase(), '').length;
                if (!keyDistance || distance < keyDistance) {
                  keyDistance = distance;
                  filtered = item;
                }
              }
            });
          }
```